### PR TITLE
bpo-46944: use FASTCALL calling convention in generator.throw

### PR DESCRIPTION
**Pull request by kumaraditya303**
https://bugs.python.org/issue46944

### CODE REVIEW

**Comment by corona10** [comment]
@kumaraditya303 Can you provide the benchmark?

**Comment by kumaraditya303** [comment]
> @kumaraditya303 Can you provide the benchmark?

Yup, I'm working on a benchmark but my machine is not very stable, if you can come up with something that would be great.

**Comment by arhadthedev** [comment]
If a vararg method looks like `throw(typ[,val[,tb]])`, can it be remolded as `throw(typ, val=None, tb=None)`?

File: Misc/NEWS.d/next/Core and Builtins/2022-03-08-10-50-42.bpo-46944.cnaIK3.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Speed up throwing exception in generator by 20%. Patch by Kumar Aditya.
```
Comment (corona10) [inline Misc/NEWS.d/next/Core and Builtins/2022-03-08-10-50-42.bpo-46944.cnaIK3.rst:1]:
```suggestion
Update .generator.throw to use FASTCALL calling convention Patch by Kumar Aditya.
```

Performance gains may be relative results depending on the environment. I prefer neutral comments.

Comment (corona10) [inline Misc/NEWS.d/next/Core and Builtins/2022-03-08-10-50-42.bpo-46944.cnaIK3.rst]: Performance gains may be relative results depending on the environment. I prefer neutral comments.
```suggestion
Update :meth:`generator.throw` to use :const:`METH_FASTCALL calling convention.  Patch by Kumar Aditya.
```

**Review by corona10 — CHANGES_REQUESTED** [review]

**Comment by kumaraditya303** [comment]
@markshannon I have added a news item, ready for your review. Thanks

**Review by corona10 — APPROVED** [review]

**Comment by markshannon** [comment]
Looks good

**Pull request merged**